hack/tilt/deploy.sh

```diff
@@ -0,0 +1,97 @@
+: ${kpng_image:=kpng}; export kpng_image
+: ${image_pull_policy:=IfNotPresent}; export image_pull_policy
+: ${backend:=nft}; export backend
+: ${config_map_name:=kpng}; export config_map_name
+: ${service_account_namee:=kpng}; export service_account_name
+: ${namespace:=kube-system}; export namespace
+: ${e2e_backend_args:=['kube', '--kubeconfig=/var/lib/kpng/kubeconfig.conf', '--exportMetrics=0.0.0.0:9099', 'to-local', 'to-nft', '--v=4', '--cluster-cidrs=fd6d:706e:6701::/56']}; export e2e_backend_args
+: ${e2e_server_args:=['kube', '--kubeconfig=/var/lib/kpng/kubeconfig.conf', '--exportMetrics=0.0.0.0:9099', 'to-local', 'to-nft', '--v=4', '--cluster-cidrs=fd6d:706e:6701::/56']}; export e2e_server_args
+: ${deployment_model:=single-process-per-node}; export deployment_model
+}
+
+function compile_bpf {
+  ###########################################################################
+  # Description:                                                            #
+  # compile bpf elf files for ebpf backend                                  #
+  ###########################################################################
+
+  pushd "${SCRIPT_DIR}"/../backends/ebpf > /dev/null || exit
+  PATH=$PATH:$BIN_DIR
+  make bytecode
+  if_error_exit "Failed to compile EBPF Programs"
+  popd > /dev/null || exit
+
+
+  info_message "Compiled BPF programs"
+}
+
+function build_kpng_and_load_image {
+  if docker build -t $KPNG_IMAGE:latest $SRC_DIR/ ; then
+	  echo "passed build"
+  else
+	  echo "failed build"
+    exit 123
+  fi
+  $BIN_DIR/kind load docker-image kpng --name kpng-proxy
+}
+
+
+if ! [[ "${backend}" =~ ^(iptables|nft|ipvs|ebpf|userspacelin|not-kpng)$ ]]; then
+  echo "skipping re-deployment due to unsupported backend"
+  exit 0 
+fi
+
+if [ "${backend}" == "ebpf" ] ; then
+  if [ "${ip_family}" != "ipv4" ] ; then
+  echo "skipping re-deployment. ebpf backend only supports ipv4"
+    exit 0
+  fi
+  compile_bpf
+fi
+
+SERVER_ARGS="'kube', '--kubeconfig=/var/lib/kpng/kubeconfig.conf', '--exportMetrics=0.0.0.0:9099', 'to-api', '--listen=unix:///k8s/proxy.sock'"
+BACKEND_ARGS="'local', '--api=${KPNG_SERVER_ADDRESS}', '--exportMetrics=0.0.0.0:9098', 'to-${backend}', '--v=${KPNG_DEBUG_LEVEL}'"
+    
+if [[ "${backend}" == "nft" ]]; then
+  case $ip_family in
+    ipv4 ) BACKEND_ARGS="$BACKEND_ARGS, '--cluster-cidrs=${CLUSTER_CIDR_V4}'" ;;
+    ipv6 ) BACKEND_ARGS="$BACKEND_ARGS, '--cluster-cidrs=${CLUSTER_CIDR_V6}'" ;;
+    dual ) BACKEND_ARGS="$BACKEND_ARGS, '--cluster-cidrs=${CLUSTER_CIDR_V4}', '--cluster-cidrs=${CLUSTER_CIDR_V6}'" ;;
+  esac
+fi
+
+BACKEND_ARGS="[$BACKEND_ARGS]"
+SERVER_ARGS="[$SERVER_ARGS]" 
+export e2e_backend_args=$BACKEND_ARGS
+export e2e_server_args=$SERVER_ARGS
+
+
+build_kpng_and_load_image
+
+rm -rf ${SCRIPT_DIR}/kpng-deployment-ds.yaml
+kubectl -n ${namespace} delete ds kpng 
+
+export_vars 
+
+go run ${SCRIPT_DIR}/kpng-ds-yaml-gen.go ${SCRIPT_DIR}/kpng-deployment-ds-template.txt ${SCRIPT_DIR}/kpng-deployment-ds.yaml
```

Comment:
at this point I'm afraid of scripts doing the same thing without reuse :(

Reply:
yes, I'll try to move common codes to a separate file